File: Array/3940.Limit Occurrences in Sorted array.java
class Solution {
    public int[] limitOccurrences(int[] nums, int k) {
        int n=nums.length;
        HashMap<Integer,Integer> mp=new HashMap<>();
        ArrayList<Integer> list=new ArrayList<>();
        for(int i=0;i<n;i++){
           mp.put(nums[i],mp.getOrDefault(nums[i],0)+1);
           if(mp.get(nums[i])<=k){
            list.add(nums[i]);
           }
        }
        int ans[]=new int[list.size()];
        for(int i=0;i<list.size();i++){
            ans[i]=list.get(i);
        }
        return ans;
    }
}

File: Array/3940.Limit Occurrences in Sorted Array.cpp
class Solution {
public:
    vector<int> limitOccurrences(vector<int>& nums, int k) {
        int n=nums.size();
        vector<int> ans;
        unordered_map<int,int> mp;
        for(int i=0;i<n;i++){
            mp[nums[i]]++;
            if(mp[nums[i]]<=k){
                ans.push_back(nums[i]);
            }
        }
        return ans;
    }
};
